sqa/tron/query.py

```diff
@@ -0,0 +1,444 @@
+    owner: list[str]
+    contract: list[str]
+    sighash: list[str]
+    logs: bool
+    internalTransactions: bool
+
+
+class InternalTxRequest(TypedDict, total=False):
+    caller: list[str]
+    transferTo: list[str]
+    transaction: bool
+
+
+class _FieldSelectionSchema(mm.Schema):
+    block = field_map_schema(BlockFieldSelection)
+    log = field_map_schema(LogFieldSelection)
+    transaction = field_map_schema(TxFieldSelection)
+    internalTransaction = field_map_schema(InternalTransactionFieldSelection)
+
+
+class _LogRequestSchema(mm.Schema):
+    address = mm.fields.List(mm.fields.Str())
+    topic0 = mm.fields.List(mm.fields.Str())
+    topic1 = mm.fields.List(mm.fields.Str())
+    topic2 = mm.fields.List(mm.fields.Str())
+    topic3 = mm.fields.List(mm.fields.Str())
+    transaction = mm.fields.Boolean()
+
+
+class _TxRequestSchema(mm.Schema):
+    type = mm.fields.List(mm.fields.Str())
+    logs = mm.fields.Boolean()
+    internalTransactions = mm.fields.Boolean()
+
+
+class _TransferTxRequestSchema(mm.Schema):
+    owner = mm.fields.List(mm.fields.Str())
+    to = mm.fields.List(mm.fields.Str())
+    logs = mm.fields.Boolean()
+    internalTransactions = mm.fields.Boolean()
+
+
+class _TransferAssetTxRequestSchema(mm.Schema):
+    owner = mm.fields.List(mm.fields.Str())
+    to = mm.fields.List(mm.fields.Str())
+    asset = mm.fields.List(mm.fields.Str())
+    logs = mm.fields.Boolean()
+    internalTransactions = mm.fields.Boolean()
+
+
+class _TriggerSmartContractTxRequestSchema(mm.Schema):
+    owner = mm.fields.List(mm.fields.Str())
+    contract = mm.fields.List(mm.fields.Str())
+    sighash = mm.fields.List(mm.fields.Str())
+    logs = mm.fields.Boolean()
+    internalTransactions = mm.fields.Boolean()
+
+
+class _InternalTransactionRequestSchema(mm.Schema):
+    caller = mm.fields.List(mm.fields.Str())
+    transferTo = mm.fields.List(mm.fields.Str())
+    transaction = mm.fields.Boolean()
+
+
+class _QuerySchema(BaseQuerySchema):
+    fields = mm.fields.Nested(_FieldSelectionSchema())
+
+    logs = mm.fields.List(mm.fields.Nested(_LogRequestSchema()))
+
+    transactions = mm.fields.List(mm.fields.Nested(_TxRequestSchema()))
+
+    transferTransactions = mm.fields.List(mm.fields.Nested(_TransferTxRequestSchema()))
+
+    transferAssetTransactions = mm.fields.List(mm.fields.Nested(_TransferAssetTxRequestSchema()))
+
+    triggerSmartContractTransactions = mm.fields.List(mm.fields.Nested(_TriggerSmartContractTxRequestSchema()))
+
+    internalTransactions = mm.fields.List(mm.fields.Nested(_InternalTransactionRequestSchema()))
+
+
+QUERY_SCHEMA = _QuerySchema()
+
+
+_blocks_table = Table(
+    name='blocks',
+    primary_key=[]
+)
+
+
+_tx_table = Table(
+    name='transactions',
+    primary_key=['transaction_index'],
+    column_weights={
+        'raw_data_hex': 'raw_data_hex_size'
+    }
+)
+
+
+_logs_table = Table(
+    name='logs',
+    primary_key=['transaction_index', 'log_index'],
```

Comment:
For logs table it is better to be compatible with EVM?

Reply:
i believe it is up to us. wouldn't it be confusing to have one entity (internal tx) indexed relatively to tx and another one (log) relatively to block?